src/rtos/rtos.cpp

```diff
@@ -96,13 +96,13 @@ using namespace pros::c;
     return task_get_count();
   }
 
-  Mutex::Mutex(void) : mutex(mutex_create()) { }
+  Mutex::Mutex(void) : mutex(std::shared_ptr<std::remove_pointer_t<mutex_t>>{mutex_create(), mutex_delete}) { }

```

Comment:
I'm pretty sure this can be simplified to:
```suggestion
  Mutex::Mutex(void) : mutex{mutex_create(), mutex_delete} { }
```

Reply:
You are very correct